### ISSUE TITLE
"Tine 2.0 needs to be updated or is not installed yet."

### ISSUE THREAD

**Issue by gtwalker** [tine20/tine20#7114]
Hi together,

currently I'm trying to install Tine 2.0 but I'm always getting this error after finishing the installation and trying to log in: "Tine 2.0 needs to be updated or is not installed yet."
I tried to install it on a Ubuntu 18.04 VM with LAMP stack, a Raspberry Pi with lighttp and mariaDB and a webspace.
Maybe I do a systematic error while configuring, but I followed the offical how to.
Can anyone help me?
This is my basic configuration (on the raspi, because its easier to work with).
![Bildschirmfoto 2019-11-25 um 23 53 46](https://user-images.githubusercontent.com/58193919/69585186-46e8b180-0fdf-11ea-9f4b-e3cd7fd27a61.png)
![Bildschirmfoto 2019-11-25 um 23 53 55](https://user-images.githubusercontent.com/58193919/69585187-46e8b180-0fdf-11ea-84a5-0e305b41c98b.png)
![Bildschirmfoto 2019-11-25 um 23 54 01](https://user-images.githubusercontent.com/58193919/69585189-46e8b180-0fdf-11ea-9bdd-992e096179b5.png)

**Comment by gerryfrancis** [comment]
The database backend is "MySQL" according to the first screenshot, but you are using MariaDB. Maybe it is possible to change the entry to "MariaDB", I do not know.

**Comment by ignoreigor** [comment]
Something went wrong - maybe you should have a look in your PHP's or webserver's error-log? Or try to activate tine 2.0's logging, there you should find errors too.

**Comment by ignoreigor** [comment]
btw: there is no special database backend for MariaDB. You are right selecting MySQL.

**Comment by gtwalker** [comment]
I don't think that might be an issue: on the ubuntu vm i have the MySQL database and MariaDB should be binary compatible with MySQL.

**Comment by lab-at-nohl** [comment]
Actually it is hard to guess. You enabled the logger, set it to DEBUG and post the log file here...

**Comment by msdotz** [comment]
Hallo an alle,
ich habe mich gerade neu angemeldet. 
Ich habe das Problem auch bei einer Neuinstallation der 2019.12.4.  Habe das auf einem Raspberry PI4 gemacht.
Während des Setups ist mir aufgefallen, das beim Setuptest bcmath und intl angemeckert wurden.
Das habe ich hingekriegt.

Dann wurde mir angezeigt, das die Konfigurationsdatei nicht schreibbar wäre.
Ist das die config.inc.php ? Die ist schreibbar, habs ausprobiert.

Dann beim ersten Login kommt die Fehlermeldung wie im Thread-Titel schon steht.
Ich bräuchte da etwas Unterstützung.
Danke

msdotz

**Comment by pschuele** [comment]
ich tippe auch auf ein problem mit der config.inc.php. kannst du schauen, ob da die richtigen db zugangsdaten drin stehen? was steht im tine log (stell mal auf DEBUG / 7)? steht was im webserver log?

**Comment by steosch** [comment]
Unfortunately I'm running into the same issue:

System: ESXi 6.7 (fully updated), Debian 10.3 Buster (fully updated, base system w/o GUI + Tine 2.0 2019.12.4); system is in a production environment since several weeks and runs perfectly smooth.

Today I updated the system

> ...# apt-get update
> ...# apt-get upgrade

and noticed that new Tine packages were available but have been kept back. After installing all Debian stable related packages I updated Tine as well 

> `...# apt-get install tine20`

which updated all Tine related packages. The output messages did not indicate any issue: 

> ...
> Conf tine20 already enabled)
> Setting up tine20 (2019.12.5-1.100
> Processing triggers for mime-support (3.62)
> ...

Apart from the updated package nothing has changed on the system. I should note that within the Tine GUI I have not observed any notifications regarding a new version being available. 

http://<ipaddress>/tine20/setup.php page can be reached and no issues to be observed there. either. Unfortunately the http://<ipaddress>/tine20 just shows "Tine 2.0 needs to be updated or is not installed yet" rendering the installation unusable. 

Fortunately I had a backup of the VM to revert to, but the latest set of install packages apparently break an existing installation. Running

> cd /usr/share/tine20
> php setup.php --update
> Database section not found in central configuration file

I'm not an expert on databases but I'm available for further inputs/help if deemed useful. Hope this helps. 

Keep up the good work!
Stefan

**Comment by lab-at-nohl** [comment]
@steosch 
> Unfortunately I'm running into the same issue:

Maybe similar but most likely different. "_Tine 2.0 needs to be updated or is not installed yet._" ist the most general error, it simply means that the program files (package) and the version of DB mismatch. 

Please Open your own issue. A Log File would be interesting...

**Comment by steosch** [comment]
... sure, no prob

Cheers, Stefan

**Comment by pschuele** [comment]
@gtwalker did you solve the problem or does it still persist? do you have a log for us?

**Comment by nebrot** [comment]
Same problem for us with a complete new installation (2020.03.1). I have debugged it in method `updateNeeded` in /Setup/Controller.php. `$this->_getUpdatesByPrio($count)` returns:
```
Array
(
    [1000] => Array
        (
            [0] => Array
                (
                    [class] => Addressbook_Setup_Update_12
                    [function] => update002
                    [prio] => 1000
                )
            [1] => Array
                (
                    [class] => Addressbook_Setup_Update_12
                    [function] => update004
                    [prio] => 1000
                )
            [2] => Array
                (
                    [class] => Addressbook_Setup_Update_12
                    [function] => update005
                    [prio] => 1000
                )
        )
)
```
Why should there a update required with a new installation? Is there a way to temporarily fix this problem?

**Comment by pschuele** [comment]
does this still happen with the new version 2020.03.2? I could not reproduce the problem. if the problem persists in the current version, please check if it also occurs if you install tine20 via CLI:

```
$ php setup.php --install
